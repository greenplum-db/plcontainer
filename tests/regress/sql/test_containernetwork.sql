\! plcontainer runtime-add -r plc_python_shared_net -i python39.alpine:latest -l python3 --setting enable_network=yes -s roles=$USER

select * from plcontainer_refresh_config;

CREATE FUNCTION access_network_err() RETURNS int AS $$
  # container: plc_python_shared
  import urllib.request
  try:
    return urllib.request.urlopen("https://www.vmware.com/").status
  except:
    return 0
$$ LANGUAGE plcontainer;

CREATE FUNCTION access_network_ok() RETURNS int AS $$
  # container: plc_python_shared_net
  import urllib.request
  return urllib.request.urlopen("https://www.vmware.com/").status
$$ LANGUAGE plcontainer;

select * from access_network_err();
select * from access_network_ok();

\! plcontainer runtime-delete -r plc_python_shared_net

-- strict permission check, except error
\! plcontainer runtime-add -r plc_python_shared_net -i python39.alpine:latest -l python3 --setting enable_network=yes
select * from plcontainer_refresh_config;
select * from access_network_ok();

drop function access_network_err();
drop function access_network_ok();

\! plcontainer runtime-delete -r plc_python_shared_net
