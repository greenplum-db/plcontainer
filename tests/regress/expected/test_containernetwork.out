\! plcontainer runtime-add -r plc_python_shared_net -i python39.alpine:latest -l python3 --setting enable_network=yes -s roles=$USER
20220718:16:25:26:178095 plcontainer:fluorite:sa-[INFO]:-Distributing file plcontainer_configuration.xml to all locations...
20220718:16:25:26:178095 plcontainer:fluorite:sa-[INFO]:-Configuration has changed. Run "select * from plcontainer_refresh_config" in open sessions. New sessions will get new configuration automatically.
select * from plcontainer_refresh_config;
 gp_segment_id | plcontainer_refresh_local_config 
---------------+----------------------------------
             1 | ok
             0 | ok
             2 | ok
            -1 | ok
(4 rows)

CREATE FUNCTION access_network_err() RETURNS int AS $$
  # container: plc_python_shared
  import urllib.request
  try:
    return urllib.request.urlopen("https://www.vmware.com/").status
  except:
    return 0
$$ LANGUAGE plcontainer;
CREATE FUNCTION access_network_ok() RETURNS int AS $$
  # container: plc_python_shared_net
  import urllib.request
  return urllib.request.urlopen("https://www.vmware.com/").status
$$ LANGUAGE plcontainer;
select * from access_network_err();
 access_network_err 
--------------------
                  0
(1 row)

select * from access_network_ok();
 access_network_ok 
-------------------
               200
(1 row)

\! plcontainer runtime-delete -r plc_python_shared_net
20220718:16:25:40:179535 plcontainer:fluorite:sa-[INFO]:-Distributing file plcontainer_configuration.xml to all locations...
20220718:16:25:40:179535 plcontainer:fluorite:sa-[INFO]:-Configuration has changed. Run "select * from plcontainer_refresh_config" in open sessions. New sessions will get new configuration automatically.
-- strict permission check, except error
\! plcontainer runtime-add -r plc_python_shared_net -i python39.alpine:latest -l python3 --setting enable_network=yes
20220718:16:25:40:179832 plcontainer:fluorite:sa-[INFO]:-Distributing file plcontainer_configuration.xml to all locations...
20220718:16:25:40:179832 plcontainer:fluorite:sa-[INFO]:-Configuration has changed. Run "select * from plcontainer_refresh_config" in open sessions. New sessions will get new configuration automatically.
select * from plcontainer_refresh_config;
 gp_segment_id | plcontainer_refresh_local_config 
---------------+----------------------------------
             2 | ok
             1 | ok
             0 | ok
            -1 | ok
(4 rows)

select * from access_network_ok();
WARNING:  plcontainer: To access network or physical device, user permission needs to be granted explicitly in the runtime config.
HINT:  set user name in <roles>
CONTEXT:  PLContainer function "access_network_ok"
ERROR:  plcontainer: Current user does not have privilege to use runtime plc_python_shared_net (plcontainer.c:343)
CONTEXT:  PLContainer function "access_network_ok"
drop function access_network_err();
drop function access_network_ok();
\! plcontainer runtime-delete -r plc_python_shared_net
20220718:16:25:41:180141 plcontainer:fluorite:sa-[INFO]:-Distributing file plcontainer_configuration.xml to all locations...
20220718:16:25:41:180141 plcontainer:fluorite:sa-[INFO]:-Configuration has changed. Run "select * from plcontainer_refresh_config" in open sessions. New sessions will get new configuration automatically.
